[ci-failure] sonobouy timing out

In jenkins-ci-ubuntu-1804-cloud-hypervisor-k8s-e2e-... for kata-containers/tests:

All the issues seem to be related to sonobouy timing out.

@cpmeadors I saw a similar fail on:

http://jenkins.katacontainers.io/job/kata-containers-2.0-ubuntu-PR-clh-containerd-k8s-minimal/290/console

But there it crash with another error:

```
20:20:41 ++ sonobuoy run --wait=180 '--e2e-focus=ConfigMap should be consumable from pods in volume|ConfigMap should be consumable from pods in volume as non-root|Kubelet when scheduling a busybox command that always fails in a pod should be possible to delete|Kubectl client Kubectl apply should reuse port when apply to an existing SVC'
20:20:41 time="2020-12-08T23:20:41Z" level=info msg="created object" name=sonobuoy namespace= resource=namespaces
20:20:41 time="2020-12-08T23:20:41Z" level=info msg="created object" name=sonobuoy-serviceaccount namespace=sonobuoy resource=serviceaccounts
20:20:41 time="2020-12-08T23:20:41Z" level=info msg="created object" name=sonobuoy-serviceaccount-sonobuoy namespace= resource=clusterrolebindings
20:20:41 time="2020-12-08T23:20:41Z" level=info msg="created object" name=sonobuoy-serviceaccount-sonobuoy namespace= resource=clusterroles
20:20:41 time="2020-12-08T23:20:41Z" level=info msg="created object" name=sonobuoy-config-cm namespace=sonobuoy resource=configmaps
20:20:41 time="2020-12-08T23:20:41Z" level=info msg="created object" name=sonobuoy-plugins-cm namespace=sonobuoy resource=configmaps
20:20:42 time="2020-12-08T23:20:42Z" level=info msg="created object" name=sonobuoy namespace=sonobuoy resource=pods
20:20:43 time="2020-12-08T23:20:43Z" level=info msg="created object" name=sonobuoy-aggregator namespace=sonobuoy resource=services
20:22:03 time="2020-12-08T23:22:03Z" level=error msg="error attempting to run sonobuoy: waiting for run to finish: Pod entered a fatal terminal status: failed"
20:22:03 + cleanup
20:22:03 ./run.sh: line 155: e2e_result_dir: unbound variable
20:22:03 Makefile:151: recipe for target 'kubernetes-e2e' failed
20:22:03 make: *** [kubernetes-e2e] Error 1
20:22:03 Failed at 50: sudo -E PATH="$PATH" CRI_RUNTIME="containerd" bash -c "make kubernetes-e2e"
```

--- a/integration/kubernetes/e2e_conformance/run.sh
+++ b/integration/kubernetes/e2e_conformance/run.sh
@@ -152,13 +152,22 @@ run_sonobuoy() {
 }
 
 cleanup() {
-	info "Results directory "${e2e_result_dir}" will not be deleted"
-	info "View results"
-	cat ${e2e_result_dir}/plugins/e2e/results/global/e2e.log
-	info "View sonobuoy status"
-	sonobuoy status
-	# Remove sonobuoy execution pods
-	sonobuoy delete
+	if [ -d "${e2e_result_dir:-}" ]; then
+		info "Results directory "${e2e_result_dir}" will not be deleted"
+		log_file="${e2e_result_dir}/plugins/e2e/results/global/e2e.log"
+		if [ -f "${log_file}" ]; then
+			info "View results"
+			cat ${log_file}
+		else
+			warn "Tests results file ${log_file} not found"
+		fi
+	fi
+	{
+		info "View sonobuoy status"
+		sonobuoy status
+		# Remove sonobuoy execution pods
+		sonobuoy delete
+	} || true
 }
 
 trap "{ cleanup; }" EXIT

